src/driver.ml

```diff
@@ -526,68 +526,79 @@ let get_whole_ast_passes ~hook ~expect_mismatch_handler ~tool_name ~input_name =
-  let exception
-    Wrapper of
-      t list
-      * label loc list
-      * (location * label) list
-      * exn
-      * Location.Error.t list
-  in
+let apply_transforms ~tool_name ~file_path ~field ~lint_field ~dropped_so_far
+    ~hook ~expect_mismatch_handler ~input_name ~embed_errors ast =
+  (* Define a custom exception to wrap the state of the transformation process
+     when an exception occurs *)
+
+  (* Get the list of AST transformations to apply *)
   let cts =
     get_whole_ast_passes ~tool_name ~hook ~expect_mismatch_handler ~input_name
   in
-  let finish (x, _dropped, lint_errors, errors) =
-    ( x,
+  (* call this function when the transformation is finished *)
+  let finish (ast, _dropped, lint_errors, errors) =
+    ( ast,
       List.map lint_errors ~f:(fun (loc, s) ->
           Common.attribute_of_warning loc s),
       errors )
   in
-  try
-    let acc =
-      List.fold_left cts ~init:(x, [], [], [])
-        ~f:(fun (x, dropped, (lint_errors : _ list), errors) (ct : Transform.t)
-           ->
-          let input_name =
-            match input_name with
-            | Some input_name -> input_name
-            | None -> "_none_"
-          in
-          let ctxt =
-            Expansion_context.Base.top_level ~tool_name ~file_path ~input_name
-          in
-          let lint_errors =
-            match lint_field ct with
-            | None -> lint_errors
-            | Some f -> (
-                try lint_errors @ f ctxt x
-                with exn when embed_errors ->
-                  raise @@ Wrapper (x, dropped, lint_errors, exn, errors))
-          in
-          match field ct with
-          | None -> (x, dropped, lint_errors, errors)
-          | Some f ->
-              let x, more_errors =
-                try f ctxt x
-                with exn when embed_errors ->
-                  raise @@ Wrapper (x, dropped, lint_errors, exn, errors)
-              in
-              let dropped =
-                if !debug_attribute_drop then (
-                  let new_dropped = dropped_so_far x in
-                  debug_dropped_attribute ct.name ~old_dropped:dropped
-                    ~new_dropped;
-                  new_dropped)
-                else []
-              in
-              (x, dropped, lint_errors, errors @ more_errors))
-    in
-    Ok (finish acc)
-  with Wrapper (x, dropped, lint_errors, exn, errors) ->
-    Error (finish (f_exception exn :: x, dropped, lint_errors, errors))
+
+  (* Attempt to apply all transformations *)
+  let exn_to_error exn =
+    match Location.Error.of_exn exn with
+    | None -> raise exn
+    | Some error -> error
+  in
+
+  let acc =
+    List.fold_left cts ~init:(ast, [], [], [])
+      ~f:(fun (ast, dropped, (lint_errors : _ list), errors) (ct : Transform.t)
+         ->
+        (* Get the input name *)
+        let input_name =
+          match input_name with
+          | Some input_name -> input_name
+          | None -> "_none_"
+        in
```

Comment:
I would say the comment is redundant with the self-explanatory code?
```suggestion
        let input_name =
          match input_name with
          | Some input_name -> input_name
          | None -> "_none_"
        in
```
That's the case for other comments!